code/modules/antagonists/traitor/objectives/sabotage_machinery.dm

```diff
@@ -0,0 +1,235 @@
+	if(!maximum_allowed)
+		return TRUE
+	if(length(possible_duplicates) >= maximum_allowed)
+		return FALSE
+	return TRUE
+
+/datum/traitor_objective/sabotage_machinery/generate_objective(datum/mind/generating_for, list/possible_duplicates)
+	var/list/possible_jobs = applicable_jobs.Copy()
+	for(var/datum/traitor_objective/sabotage_machinery/objective as anything in possible_duplicates)
+		possible_jobs -= objective.chosen_job
+	for(var/available_job in possible_jobs)
+		var/job_machine_path = possible_jobs[available_job]
+		if (!length(GLOB.objective_machine_handler.machine_instances_by_path[job_machine_path]))
+			possible_jobs -= available_job
+	if(!length(possible_jobs))
+		return FALSE
+
+	chosen_job = pick(possible_jobs)
+	var/list/obj/machinery/possible_machines = GLOB.objective_machine_handler.machine_instances_by_path[possible_jobs[chosen_job]]
+	for(var/obj/machinery/machine as anything in possible_machines)
+		prepare_machine(machine)
+
+	replace_in_name("%JOB%", lowertext(chosen_job))
+	replace_in_name("%MACHINE%", possible_machines[1].name)
+	return TRUE
+
+/// Marks a given machine as our target
+/datum/traitor_objective/sabotage_machinery/proc/prepare_machine(obj/machinery/machine)
+	AddComponent(/datum/component/traitor_objective_register, machine, succeed_signals = list(COMSIG_PARENT_QDELETING))
+
+/datum/traitor_objective/sabotage_machinery/destroy
+	name = "Destroy the %MACHINE%"
+	description = "Destroy the %MACHINE% to cause disarray and disrupt the operations of the %JOB%'s department."
+
+	progression_reward = list(5 MINUTES, 10 MINUTES)
+	telecrystal_reward = list(3, 4)
+
+	progression_minimum = 15 MINUTES
+	progression_maximum = 30 MINUTES
+
+	applicable_jobs = list(
+		JOB_STATION_ENGINEER = /obj/machinery/telecomms/hub,
+		JOB_SCIENTIST = /obj/machinery/rnd/server,
+	)
+
+/datum/traitor_objective/sabotage_machinery/trap
+	name = "Sabotage the %MACHINE%"
+	description = "Destroy the %MACHINE% to cause disarray and disrupt the operations of the %JOB%'s department. If you can have another crew member destroy the machine using the provided booby trap, you will be rewarded with an additional %PROGRESSION% reputation and %TC% telecrystal."
+
+	maximum_allowed = 2
+	applicable_jobs = list(
+		JOB_CHIEF_ENGINEER = /obj/machinery/rnd/production/protolathe/department/engineering,
+		JOB_CHIEF_MEDICAL_OFFICER = /obj/machinery/rnd/production/techfab/department/medical,
+		JOB_HEAD_OF_PERSONNEL = /obj/machinery/rnd/production/techfab/department/service,
+		JOB_QUARTERMASTER = /obj/machinery/rnd/production/techfab/department/cargo,
+		JOB_RESEARCH_DIRECTOR = /obj/machinery/rnd/production/protolathe/department/science,
+		JOB_SHAFT_MINER = /obj/machinery/mineral/ore_redemption,
+	)
+
+	/// Bonus reward to grant if you booby trap successfully
+	var/bonus_tc = 1
+	/// Bonus reputation to grant if you booby trap successfully
+	var/bonus_rep = 2 MINUTES
```

Comment:
I don't mind beefing up the bonus tc, to 2-3 and then reducing the actual awarded TC